lib/prometheus/que_stats.rb

```diff
@@ -17,19 +17,48 @@ def worker_stats
       end
     end
 
-    def job_stats(filter = nil)
-      filter = "WHERE #{filter}" if filter
+    # ApplicationJob did not have to be here, but it's just harder to test otherwise because of ApplicationJob#delete_duplicates
+    JOB_CLASSES = %w[ApplicationJob ProcessEntryJob ProcessIntegrationEntryJob UpdateJob].inspect.tr('"', "'").freeze
+
+    def job_stats(*filters)
+      filter = "WHERE #{filters.join(' AND ')}" if filters.presence
       sql = <<~SQL
-        SELECT args->0->>'job_class' AS job_class, COUNT(*) as count
-        FROM que_jobs #{filter}
-        GROUP BY args->0->>'job_class'
+        WITH
```

Comment:
Common Table expression can be done in Arel, there is an open PR in https://github.com/rails/rails/pull/37944
I think you can do it without raw SQL